popsycle/synthetic.py

```diff
@@ -3515,32 +3668,400 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+
+    return ubv_to_ztf_grid, kdtree
+
+
+def transform_ubv_to_ztf(ubv_b, ubv_v, ubv_r):
+    """
+    Converts ubv filters (b, v, r) into ztf filters (g, r)
+
+    Parameters
+    ----------
+    ubv_b : array of floats
+        ubv_b photometry of galaxia / PyPopStar sources
+
+    ubv_v : array of floats
+        ubv_v photometry of galaxia / PyPopStar sources
+
+    ubv_r : array of floats
+        ubv_r photometry of galaxia / PyPopStar sources
+
+    Output
+    ------
+    ztf_g : array of floats
+        ztf_g photometry of galaxia / PyPopStar sources
+
+    ztf_r : array of floats
+        ztf_r photometry of galaxia / PyPopStar sources
+
+    """
+
+    # Convert the ubv photometry into the right format
+    x_data = ubv_v - ubv_r
+    y_data = ubv_b - ubv_v
+    data = np.squeeze(np.dstack([x_data, y_data]), axis=0)
+
+    # Only query on data that is luminous
+    cond_lum = ~np.isnan(data).any(axis=1)
+
+    for filter_name in ['g', 'r']:
+        # Start with an empty array of nans
+        ztf_diff = np.full(len(ubv_b), np.nan)
+
+        # Find locations on the grid where x_data and y_data are located.
+        # Put those values into the ztf_diff array
+        ubv_to_ztf_grid, kdtree = load_ubv_to_ztf_grid(filter_name)
+        _, indexes = kdtree.query(data[cond_lum])
+        ztf_diff[cond_lum] = ubv_to_ztf_grid.flatten()[indexes]
+
+        # Convert to ztf_g and ztf_r
+        if filter_name == 'g':
+            ztf_g = ubv_v - ztf_diff
+        elif filter_name == 'r':
+            ztf_r = ubv_r - ztf_diff
+
+    return ztf_g, ztf_r
+
+
+def ztf_mag_vega_to_AB(ztf_mag_vega, filter_name):
+    """
+    Converts vega magnitudes into AB magnitudes for ztf filters.
+
+    Parameters
+    ----------
+    ztf_mag_vega : float, array of floats
+        ztf photometry of galaxia / PyPopStar sources in vega system
+
+    filter_name : str
+        The name of the filter in which to calculate all the
+        microlensing events. Must be either 'g' or 'r'.
+
+    Output
+    ------
+    ztf_mag_AB : float, array of floats
+        ztf photometry of galaxia / PyPopStar sources in AB system
+
+    """
+    if filter_name == 'g':
+        ztf_mag_AB = ztf_mag_vega - 0.07
+    elif filter_name == 'r':
+        ztf_mag_AB = ztf_mag_vega + 0.19
+    else:
+        print('filter_name must be either g or r')
+        ztf_mag_AB = None
+    return ztf_mag_AB
+
+
+def ztf_mag_AB_to_vega(ztf_mag_AB, filter_name):
+    """
+    Converts AB magnitudes into vega magnitudes for ztf filters.
+
+    Parameters
+    ----------
+    ztf_mag_AB : float, array of floats
+        ztf photometry of galaxia / PyPopStar sources in AB system
+
+    filter_name : str
+        The name of the filter in which to calculate all the
+        microlensing events. Must be either 'g' or 'r'.
+
+    Output
+    ------
+    ztf_mag_vega : float, array of floats
+        ztf photometry of galaxia / PyPopStar sources in vega system
+
+    """
+    if filter_name == 'g':
```

Comment:
Provide a reference for where these hard-coded numbers come from.

Reply:
Done